### PR TITLE
Bugfix Unified Search Load More

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

In certain scenarios even user have more data, "Load More Data" not visible.

e.g. 

User have F1, F2, F3, F4, f 5, f 6, f7, f 8 folders when user make search via "f" user only sees last 4 item without "load more data"

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt [hunk 1]
```diff
@@ -94,14 +94,6 @@ class UnifiedSearchViewModel(application: Application) : AndroidViewModel(applic
         )
     }
 
-    open fun startLoading(query: String) {
```
Comment (alperozturk96) [inline app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt:97]: unused function

File: app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt [hunk 2]
```diff
@@ -185,10 +177,6 @@ class UnifiedSearchViewModel(application: Application) : AndroidViewModel(applic
         }
     }
 
-    open fun clearError() {
```
Comment (alperozturk96) [inline app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt:188]: unused function

File: app/src/main/java/com/owncloud/android/ui/unifiedsearch/SearchOnProviderTask.kt [hunk 3]
```diff
@@ -16,7 +16,7 @@ class SearchOnProviderTask(
     private val provider: String,
     private val client: NextcloudClient,
     private val cursor: Int? = null,
-    private val limit: Int = 5
+    private val limit: Int = 6
```
Comment (alperozturk96) [inline app/src/main/java/com/owncloud/android/ui/unifiedsearch/SearchOnProviderTask.kt:19]: Why is the result of UnifiedSearchRemoteOperation 1 less than the limit we have? If limit is 6 I'm getting 5 results, if 5 I'm getting 4 results etc.

File: app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt [hunk 4]
```diff
@@ -213,11 +194,13 @@ class UnifiedSearchViewModel(application: Application) : AndroidViewModel(applic
...
             .filter { it.value.results.isNotEmpty() }
             .map { (key, value) ->
+                val hasMoreResults = results.isNotEmpty() && results[key]?.nextCursor() != null
+
                 UnifiedSearchSection(
                     providerID = key,
                     name = value.name()!!,
                     entries = value.results.flatMap { it.entries },
-                    hasMoreResults = !value.isFinished()
+                    hasMoreResults = hasMoreResults
```
Comment (AndyScherzinger) [inline app/src/main/java/com/owncloud/android/ui/unifiedsearch/UnifiedSearchViewModel.kt:203]: This statement seems not needed...

**Review by tobiasKaminsky — APPROVED** [review]

**Pull request merged**